kernel/mem_domain.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <kernel.h>
+#include <kernel_structs.h>
+#if defined(CONFIG_ARM_CORE_MPU)
+#include <arch/arm/cortex_m/mpu/arm_core_mpu.h>
+#endif
+
+
+static u8_t max_partitions;
+
+
+static inline u8_t arch_get_max_partitions(void)
+{
+#if defined(CONFIG_ARM_CORE_MPU)
```

Comment:
Shouldn't these `arch_*` functions be in arch-specific headers?

Reply:
yeah Chunlin is going to revise this, we discussed in PR #814